runtime/collections/rangemaps.cpp

```diff
@@ -0,0 +1,282 @@
+#include "runtime/header.h"
+
+#include "immer/flex_vector_transient.hpp"
+
+extern "C" {
+rangemap
+hook_RANGEMAP_element(SortKItem start, SortKItem end, SortKItem value) {
+  return rangemap().inserted(rng_map::Range<KElem>(start, end), value);
+}
+
+struct range {
+  blockheader h;
+  block *start;
+  block *end;
+};
+static struct blockheader range_header() {
+  static struct blockheader hdr = {(uint64_t)-1};
+  if (hdr.hdr == -1) {
+    hdr = getBlockHeaderForSymbol(
+        (uint64_t)getTagForSymbolName("LblRangemap'Coln'Range{}"));
+  }
+  return hdr;
+}
+rangemap hook_RANGEMAP_elementRng(SortKItem rng, SortKItem value) {
+  range *ptr = (range *)rng;
```

Comment:
This doesn't seem quite right. You marked rng as being of sort KItem, but then you cast it to `range *`, when I'm pretty sure you also explicitly created a `Range` type. Either `rng` needs to be of sort `Range`, or else you're going to need to deconstruct an injection here. It depends on what the frontend code is expecting in terms of a type.

Reply:
Addressed, using sort Range.